### PR TITLE
Adds support for external SPGs to net-firewall-policy

### PR DESCRIPTION
**Pull request by LucaPrete**
- Adds support to reference external SPGs in factories in net-firewall-policy
- Adds missing `tls_inspect` argument to hierarchical and global network firewall policies
- Fixes regional firewall policy rules, removing security profile groups and ngfw actions (given they're not supported)
- Updates copyright

**Checklist**

I applicable, I acknowledge that I have:
- [X] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [X] Ran `terraform fmt` on all modified files
- [X] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [X] Made sure all relevant tests pass

### CODE REVIEW

File: modules/net-firewall-policy/README.md [hunk 1]
```diff
@@ -253,19 +253,77 @@ issue-1995:
       - 1-65535
     - protocol: icmp
 ```
+
+You might need to reference external security profile groups in your firewall rules, using thier Terraform ids. For example, `//networksecurity.googleapis.com/${google_network_security_security_profile_group.security_profile_group.id}`. To do so, list your security profile groups in the `security_profile_group_ids` map variable. Then reference them by key from your factories.
```
Comment (ludoo) [inline modules/net-firewall-policy/README.md:257]: thier ---> their

File: modules/net-firewall-policy/README.md [hunk 2]
```diff
@@ -253,19 +253,77 @@ issue-1995:
...
+  location                  = "global"
+  description               = "Sample security profile group."
+  threat_prevention_profile = google_network_security_security_profile.security_profile.id
+}
+
+module "firewall-policy" {
+  source    = "./fabric/modules/net-firewall-policy"
+  name      = "fw-policy"
+  parent_id = "my-project"
+
```
Comment (ludoo) [inline modules/net-firewall-policy/README.md:285]: remove newline

File: modules/net-firewall-policy/net-global.tf [hunk 3]
```diff
@@ -44,11 +44,16 @@ resource "google_compute_network_firewall_policy_rule" "net-global" {
...
   description             = local.rules[each.key].description
   direction               = local.rules[each.key].direction
-  security_profile_group  = local.rules[each.key].security_profile_group
   disabled                = local.rules[each.key].disabled
   enable_logging          = local.rules[each.key].enable_logging
   priority                = local.rules[each.key].priority
   target_service_accounts = local.rules[each.key].target_service_accounts
+  tls_inspect             = local.rules[each.key].tls_inspect
+  security_profile_group = (
+    try(var.security_profile_group_ids[local.rules[each.key].security_profile_group], null) == null
```
Comment (ludoo) [inline modules/net-firewall-policy/net-global.tf:53]:
```hcl
try(
  var.security_profile_group_ids[local.rules[each.key].security_profile_group],
  local.rules[each.key].security_profile_group
)
```

or lookup which is better than try

```hcl
lookup(
  var.security_profile_group_ids,
  local.rules[each.key].security_profile_group,
  local.rules[each.key].security_profile_group
)
```

File: modules/net-firewall-policy/hierarchical.tf [hunk 4]
```diff
@@ -37,12 +37,17 @@ resource "google_compute_firewall_policy_rule" "hierarchical" {
...
   enable_logging          = local.rules[each.key].enable_logging
   priority                = local.rules[each.key].priority
   target_resources        = local.rules[each.key].target_resources
   target_service_accounts = local.rules[each.key].target_service_accounts
+  tls_inspect             = local.rules[each.key].tls_inspect
+  security_profile_group = (
+    try(var.security_profile_group_ids[local.rules[each.key].security_profile_group], null) == null
+    ? local.rules[each.key].security_profile_group
+    : var.security_profile_group_ids[local.rules[each.key].security_profile_group]
+  )
```
Comment (ludoo) [inline modules/net-firewall-policy/hierarchical.tf:50]: same comment as for global

**Review by ludoo — APPROVED** [review]
one typo and I would redo the try/lookup

File: modules/net-firewall-policy/net-global.tf (hunk 3 above)
Comment (LucaPrete) [reply modules/net-firewall-policy/net-global.tf:53]: I tried lookup before and it doesn't work as the second argument (the lookup key) can never be null, which happens instead in this case. I'll use the try.

**Pull request merged**